rcldotnet/rcldotnet.c

```diff
@@ -46,6 +53,16 @@ void native_rcl_reset_error() { rcl_reset_error(); }
 bool native_rcl_ok() { return rcl_ok(); }
 
 int32_t native_rcl_create_node_handle(void **node_handle, const char *name, const char *namespace) {
+
+  // Validate the node name
+  int validation_result;
+  rmw_ret_t rmw_ret =
+    rmw_validate_node_name(name, &validation_result, NULL);
+
+  if (rmw_ret != RMW_RET_OK || validation_result != RMW_NODE_NAME_VALID) {
```

Comment:
This can be simplified to just return `rmw_ret` if it's not `RMW_RET_OK`

Reply:
Actually, the return value is validation_result. 

Here there is an issue because this function can return returning value of  `rmw_validate_node_name` and `rcl_node_init`. I tried to take into account errors from both functions.